book/content/material/material_basics.md

```diff
@@ -0,0 +1,198 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.17.3
+kernelspec:
+  display_name: festim-workshop
+  language: python
+  name: python3
+---
+
+# Basic Material Features #
+
+Materials are key components of hydrogen transport simulations. They hold the properties like diffusivity, solubility and even thermal properties like thermal conductivity or heat capacity. Read more about the `Materials` class and syntax at __[Materials](https://festim.readthedocs.io/en/v1.0.1/userguide/materials.html)__.
+
+
+Objectives:
+* Learn how to define material properties (thermal, solubility, diffusion)
+* Learn how to define materials on different subdomains
+* Solve a multi-material diffusion problem
+
++++
+
+## Defining material properties ##
+
+We can define a material using the diffusion exponential pre-factor $D_0$ and activation energy $E_d$, which is material dependent:
+
+```{code-cell} ipython3
+import festim as F
+
+mat = F.Material(D_0=1.11e-6, E_D=0.4)  # m2/s, eV
+```
+
+When considering chemical potential conservation at material interfaces, hydrogen solubility can be defined using the solubility coeffeicient prefactor `K_S_0`, solubility activation energy `E_K_S`, and solubility law `solubility_law` (either `"henry"` or `"sievert"`):
+
+```{code-cell} ipython3
+mat.K_S_0 = 1.0
+mat.E_K_S = 3.0
+mat.solubility_law = "sievert"
+```
+
+Some material properties are temperature-dependent. Users can define such thermal properties (thermal conductivity, heat capacity, density) as function of temperature in the following way:
+
+```{code-cell} ipython3
+import ufl
+
+mat.thermal_conductivity = lambda T: 3*T + 2*ufl.exp(-20*T)
+mat.heat_capacity = lambda T: 4*T + 8
+mat.density = lambda T: 7*T + 5
+```
+
+## Defining materials on different subdomains ##
+
+Volume subdomains are used to assign different materials or define regions with specific physical properties. Each volume subdomain must be associated with a `festim.Material` object. Read more about subdomains __[here](https://festim-workshop.readthedocs.io/en/festim2/content/meshes/mesh_fenics.html)__.
+
+Consider the following volume with two subdomains separated halfway through the mesh:
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+import festim as F
+from dolfinx.mesh import create_unit_square
+from mpi4py import MPI
+
+fenics_mesh = create_unit_square(MPI.COMM_WORLD, 10, 10)
+festim_mesh = F.Mesh(fenics_mesh)
+```
```

Comment:
This is weird because one doesn't need either `fenics_mesh` or `festim_mesh` to define different subdomains with the same material
```suggestion
```